### DISCUSSION TITLE
SONOFF TRVZB zigbee support

### DISCUSSION THREAD

**Discussion by kelsotowers** [arendst/Tasmota#20568] [Support request]
Hi, my first support request on here. I've been using Tasmota using wifi on switches and H&T devices for a while but I'm trying to get a couple of different zigbee TRVs working. My Hive (Danfoss) device is working better than the SONOFF TRVZB and I wonder if anyone else has got this device working properly?

I'm using Tasmota 13.3.0.2(zbbrdgpro) on a Sonoff Zigbee Bridge Pro  with ZNP v2.7.1.20230507.

The Hive TRV reports the local temperature and set point temperature; and the valve opening % from PIHeatingDemand as it should. But the SONOFF TRVZB only reports the local temperature and the set point temperature. The valve opening isn't reported, just an additional line showing 'On' - as per the screenshot. There are other parameters I can't read but I think this is a good example.

![IMG_8384](https://github.com/arendst/Tasmota/assets/11380351/e3098f06-bd98-4ce5-8810-08710888586e)

Has anyone successfully controlled a SONOFF TRVZB via Tasmota? It seems to work better via Zigbee2MQTT to be honest, but I'm much rather use Tasmota like the rest of my devices.

Thanks to anyone that reads this!

**Comment by s-hadinger** [comment]
I don't know this specific device. Do you have the detailed logging when you paired it? If not can tou pair it again with logs enabled (Weblog 3)

**Reply by kelsotowers** [reply]
Thank you for replying. I've deleted and rejoined the device and also removed my own attempt at a .zb configuration file so hopefully there will be nothing in here to confuse it from a virgin attempt. I attach the log - I've copied and pasted this from the console, if there is a better way to capture the log please let me know. I appreciate you looking at this, thank you!
[TRVZB_pairing.pdf](https://github.com/arendst/Tasmota/files/14024507/TRVZB_pairing.pdf)

**Comment by jeroenst** [comment]
Is this sonoff trvzb now working correctly with tasmota? I want to buy it if it works fine.

**Comment by kelsotowers** [comment]
I have one that I flashed with Tasmota and I can set the temperature via mosquito. It’s in my workshop which I don’t heat regularly but when I want to, it works fine. I haven’t explored the finer points - I just set it to 6C to keep the frost out or 20C when I go in there to work. It does that reliably and reports battery and link quality regularly too. On 19 Jan 2025, at 15:02, JeroenSt ***@***.***> wrote:﻿
Is this sonoff trvzb now working correctly with tasmota? I want to buy it if it works fine.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/arendst/Tasmota/discussions/20568#discussioncomment-11882011",
"url": "https://github.com/arendst/Tasmota/discussions/20568#discussioncomment-11882011",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by jeroenst** [reply]
Thank you, does it also report temperature and temperature setpoint and maybe the percentage of valve opening?

**Comment by kelsotowers** [comment]
I don’t read % open - just ‘on’ or ‘off’. This is what I publish on my house web page. Surprisingly 0% link quality works fine!On 19 Jan 2025, at 15:36, JeroenSt ***@***.***> wrote:﻿
Thank you, does it also report temperature and temperature setpoint and maybe the percentage of valve opening?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/arendst/Tasmota/discussions/20568#discussioncomment-11882246",
"url": "https://github.com/arendst/Tasmota/discussions/20568#discussioncomment-11882246",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
